src/Backend/AMQPBackend.php

```diff
@@ -209,7 +209,11 @@ public function getIterator()
         $this->consumer = $this->getContext()->createConsumer($this->getContext()->createQueue($this->queue));
         $this->consumer->setConsumerTag('sonata_notification_'.uniqid());
 
-        return new AMQPMessageIterator($this->getChannel(), $this->consumer);
+        if (!$context instanceof \Enqueue\AmqpLib\AmqpContext) {
+            throw new \LogicException('The BC layer works only if enqueue/amqp-lib lib is being used.');
```

Comment:
I guess it throws some kind of exception already? else this is a BC break

Reply:
It is not as 3.x version works only with `Enqueue\AmqpLib`. It replaces php-amqplib. 4.x version could work with any amqp interop compatible transport.